### PR TITLE
Add OpenGraph meta tags to support link previews

### PR DESCRIPTION
**Pull request by aabounegm**

### CODE REVIEW

File: src/routes/projects/past.svelte [hunk 1]
```diff
@@ -43,6 +43,9 @@
 
 <svelte:head>
   <title>Projects – Innopoints</title>
+  <meta name="og:title" content="Past Projects" />
+  <meta name="og:url" content="https://ipts.innopolis.university/projects/past" />
+  <meta name="og:description" content="Volunteering opportunities that are already over" />
```
Comment (illright) [inline src/routes/projects/past.svelte:48]: How about _A history of volunteering opportunities at Innopolis University_

File: src/routes/projects/index.svelte [hunk 2]
```diff
@@ -71,6 +71,9 @@
 
 <svelte:head>
   <title>Projects – Innopoints</title>
+  <meta name="og:title" content="Projects" />
+  <meta name="og:url" content="https://ipts.innopolis.university/projects" />
+  <meta name="og:description" content="If you’re seeking an opportunity to contribute to the wonderful world of Innopolis projects, you’re in the right place" />
```
Comment (illright) [inline src/routes/projects/index.svelte:76]: Period.

File: src/routes/projects/[id([0-9]+)]/index.svelte [hunk 3]
```diff
@@ -355,6 +356,12 @@
 
 <svelte:head>
   <title>{project.name} – Innopoints</title>
+  <meta name="og:title" content={project.name} />
+  <meta name="og:url" content="https://ipts.innopolis.university/projects/{project.id}" />
+  <meta name="og:description" content="Organized by {project.organizer}" />
```
Comment (illright) [inline src/routes/projects/[id([0-9]+)]/index.svelte:361]: Pretty sure we can fit in another line saying: _Available activities: X, Y, Z..._, at most three, less if we run out of space

File: src/routes/products/[id([0-9]+)]/index.svelte [hunk 4]
```diff
@@ -106,6 +107,12 @@
 
 <svelte:head>
   <title>{fullName} – Innopoints</title>
+  <meta name="og:title" content={fullName} />
+  <meta name="og:url" content="https://ipts.innopolis.university/products/{product.id}" />
+  <meta name="og:description" content={product.description} />
```
[the remark this answers is not in the document]
Comment (illright) [reply src/routes/products/[id([0-9]+)]/index.svelte:112]: Let's put a placeholder like _Yet another cool item that could be yours!_

File: src/routes/index.svelte [hunk 5]
```diff
@@ -27,6 +27,10 @@
 
 <svelte:head>
   <title>Home – Innopoints</title>
+  <meta name="og:title" content="Home" />
+  <meta name="og:url" content="https://ipts.innopolis.university" />
+  <meta name="og:description" content="Motivating the great beginnings" />
```
[the remark this answers is not in the document]
Comment (illright) [reply src/routes/index.svelte:32]: Delegated this to Andrey, he said he'll _do it later_. As of now, we can use this one and update later on

[the remark this answers is not in the document]
Comment (illright) [reply src/routes/index.svelte:32]: _Innopoints is a system that encourages extracurricular participation. It helps to manage volunteers, track soft skills, obtain feedback and get branded Innopolis University merchandise as a reward!_

File: src/routes/index.svelte [hunk 6]
```diff
@@ -27,6 +27,10 @@
 
 <svelte:head>
   <title>Home – Innopoints</title>
+  <meta name="og:title" content="Home" />
```
[the remark this answers is not in the document]
Comment (aabounegm) [reply src/routes/index.svelte:30]: Quoting that article, "But for our purposes, and for the sake of brevity, we can leverage the fact that Twitter allows us to substitute Open Graph <meta> tags for its own. In the end, except for the necessity of specifying a display format, none of the custom Twitter <meta> tags are needed."
If you were referring just to `<meta name="twitter:card" content="summary_large_image">`, I will add it, even though it doesn't seem to make a difference. This is how it looks like on https://cards-dev.twitter.com/validator, both with and without that tag:
image

File: src/template.html [hunk 7]
```diff
@@ -1,9 +1,14 @@
 <!DOCTYPE html>
-<html lang="en">
+<html lang="en" prefix="og: http://ogp.me/ns#">
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width,initial-scale=1.0" />
     <meta name="theme-color" content="#127800" />
+    <meta property="og:site_name" content="Innopoints" />
+    <meta property="og:type" content="university" />
```
[the remark this answers is not in the document]
Comment (aabounegm) [reply src/template.html:8]: I'm sure I saw it listed in one of the articles I referred to, but I cannot find it now and it is not in the official list, so I guess you're right

File: src/routes/products/[id([0-9]+)]/index.svelte [hunk 8]
```diff
@@ -106,6 +107,12 @@
...
+  {#if productControl.flatImages.length > 0}
+    <meta name="og:image" content="{API_HOST_BROWSER + productControl.flatImages[0].url}" />
```
[the remark this answers is not in the document]
Comment (aabounegm) [reply src/routes/products/[id([0-9]+)]/index.svelte:114]: For some reason, [Facebook's tool](https://developers.facebook.com/tools/debug/) couldn't fetch the Ngrok-tunnelled website, Twitter's tool didn't show any image (as in the screenshot in a previous comment), and metatags.io doesn't seem to show any tags (even though it correctly extracted and printed them to the console). I would say what matters now is that it is working on Telegram.
Additionally, we don't even have a 2:1 image representing the website, and I wouldn't like one that is just filled with whitespace. A favicon shouldn't act as the link preview image anyways

**Pull request merged**